spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -401,22 +401,17 @@ def __init__(
             n_clu = clu[0]
             clu = np.delete(clu, 0)
             unique_ids = np.unique(clu)
-            if 0 not in unique_ids:  # missing unsorted IDs
-                n_clu += 1
-            if 1 not in unique_ids:  # missing mua IDs
-                n_clu += 1
+            assert len(unique_ids) == n_clu, (
+                "First value of .clu file ({clufile_path}) does not match number of unique IDs!"
+            )
+            unit_map = dict(zip(unique_ids, list(range(n_clu))))
```

Comment:
If it is possible possible to have an empty unit ( a unit with no spike) then this check could lead to problem.
In neo for instance it is possible to have unit without spikes.
In short this `unique_ids = np.unique(clu)` do not garanty to have the full unit list.

Reply:
This assertion is simply that the Neuroscope file is correct to its own format. The `n_clu` value is the first line of what is essentially a `.txt` file, and _should_ indicate the number of units reported in the file (including unsorted and mua). The rest of the values in the `.txt` file are the unit IDs for each spike occurrence (one for each spike occurrence, in order of time of occurrence - that info, though, is in a separate file).

For Neuroscope format it's therefore not possible to have a unit with no spikes, since as we're learning now the actual ID of each non-trivial unit is arbitrary (not incremented from 2, 3, 4, ... as previously thought). Note the trivial unit IDs of 0 (unsorted, always ignored) and 1 (mua, optional to include) are not arbitrary.

What we'd really need for both this and neo is some sample data for the format on GIN - I think that's been on Julia's to-do list for a while but not sure what the status is on it.